Create README.md

Bartmax opened the pull request:
First iteration. Modifications and suggestions are welcome.
fix for https://github.com/openiddict/core/issues/1

Commit:
Create README.md
First iteration. Modifications and suggestions are welcome.

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,194 @@
+# OpenIddict, as an OpenID Connect Addict
+
+### What's Open Iddict ?
```
nit: OpenIddict.

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,194 @@
+server for ASP.NET 5.
+OpenIddict leverages the use of Identity (for user management) and Entityframework (as store provider).
+
+### What's Open ID Connect ?
```
nit: OpenID Connect.

ilmax replied:
nit: link to OpenID connect site/documentation

Bartmax replied:
Great point @ilmax ! Should we mention ASOS too and link to it ?

ilmax replied:
Sure :)

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,194 @@
+
+### What's Open ID Connect ?
+
+Open ID is a protocol that allows multiple clients to delegate the Authentication to one or multiple Authority servers.
```
nit: `Open ID` -> `OpenID Connect` (OpenID and OpenID Connect are two different specs)

kevinchalet replied:
nit: `the Authentication` -> `authentication` (or `the authentication process`)

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,194 @@
+Open ID is a protocol that allows multiple clients to delegate the Authentication to one or multiple Authority servers.
+Facebook, Google, Twitter, are all well-known as OpenId Authority Servers.
+
+### Why should I want/need my own Open ID Connect Server?
```
nit: OpenID Connect server

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,194 @@
+There are many scenarios on when this may be useful, 
+
+ 1. Having multiple applications with one central place for authentication.
+ 2. Having non-trusted clients (like JavaScript, Desktop or Mobile applications).
```
Actually, the client are not necessarily non-trusted. You can have totally trusted MVC clients.
I guess you meant external clients, where you want to allow third-party developers to access your users' data?

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,194 @@
+
+ 1. Having multiple applications with one central place for authentication.
+ 2. Having non-trusted clients (like JavaScript, Desktop or Mobile applications).
+ 3. Needs or wants to use single sign-on for multiple services.
```
Sounds like the same as the first one, no?

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,194 @@
+ 2. Having non-trusted clients (like JavaScript, Desktop or Mobile applications).
+ 3. Needs or wants to use single sign-on for multiple services.
+
+In general terms, you want to have a centralized authentication server/login and then any allowed clients
```
This sentence sounds a bit... weird? :smile:

kevinchalet commented on README.md:
````diff
@@ -0,0 +1,194 @@
+    },
+```
+
+In ConfigureServices there's a handy extension method for IdentityBuilder that you need to set up.
````
nit: consider using backticks for types/methods/properties: ConfigureServices -> `ConfigureServices`

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,194 @@
+
+### Configuration & Options.
+
+<p>OpenIddict have multiple options you can set to customize for your requirements/needs.<p>
```
Has?

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,194 @@
+  <td>Issuer</td>
+  <td>Uri</td>
+  <td>The base address (absolute) used to uniquely identify the authorization server.<br>
+      <sub><i>Unless <code>AllowInsecureHttp</code> has been set to <code>true</code>, an HTTPS address must be provided.</i></sub>
```
I should probably fix ASOS' documentation to mention this property is now optional. When left to null, the issuer property is automatically inferred from the request URL.

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,194 @@
+
+## Contributors
+
+**OpenIddict** is actively maintained by **[Kévin Chalet](https://github.com/PinpointTownes)**. Contributions are welcome and can be submitted using pull requests.
```
Feel free to add your name! :clap:

ilmax commented on README.md:
```diff
@@ -0,0 +1,194 @@
+        .AddOpenIddict();
+
+    /// .... other services you may have
+}
```
missing closing ```

Commit:
Updated README.md

Improved description of general concepts and benefits of having an
OpenID Connect server.
Added reference to ASOS and OpenID Connect website.
Fixed typos, grammar and notation in general.

Commit:
Minor update README.md

fixed missing backticks `.

kevinchalet commented:
Should we also indicate that nightly builds can now be found on the aspnet-contrib MyGet repository? https://www.myget.org/F/aspnet-contrib/api/v3/index.json

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,224 @@
+OpenIddict library aims to provide a simple and easy out of the box solution 
+to implement an OpenID Connect server for ASP.NET 5.
+
+It does leverages the use of Identity (for user management) and 
```
nit: `does leverages` -> `leverages`.

Commit:
Added mention to nightly builds repository.

ilmax commented on README.md:
```diff
@@ -0,0 +1,226 @@
+Provider (IdP).
+
+OpenID Connect is a standard protocol on top of OAuth that enables Clients to 
+verify the identity of the End-User based on the authentication performed by 
```
on top of OAuth 2.0 protocol

Commit:
Fix OAuth 2.0 spelling/version

OAuth --> OAuth 2.0

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,226 @@
+</tbody>
+</table>
+
+You can find working samples in the [samples](https://github.com/openiddict/core/tree/dev/samples) directory.
```
I think we should move these 2 paragraphs in the `Getting started` section.

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,226 @@
+
+You can find working samples in the [samples](https://github.com/openiddict/core/tree/dev/samples) directory.
+
+Nightly builds can now be found on the [aspnet-contrib](https://github.com/aspnet-contrib) MyGet repository https://www.myget.org/F/aspnet-contrib/api/v3/index.json.
```
Yeah, move this one at the very begging of the `Getting started` section :smile:

Commit:
README.md update

Working samples and nightly builds on top of gettings started section.

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,226 @@
+  <td><code>false</code></td>
+</tr>
+<tr>
+  <td>UseCustomViews</td>
```
FYI: this option will be removed when https://github.com/openiddict/core/pull/25 is merged :warning:

Bartmax replied:
:+1: :watch:

Commit:
README.md react to https://github.com/openiddict/core/pull/25

Removed UseCustomViews

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,212 @@
+# OpenIddict, as an OpenID Connect Addict
```
OpenIddict, the OpenID Connect server you'll be addict to :smile:

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,212 @@
+
+### What's OpenIddict ?
+
+OpenIddict library aims to provide a simple and easy out of the box solution 
```
nit: `OpenIddict library aims to provide` -> `OpenIddict aims at providing`

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,212 @@
+It leverages the use of Identity (for user management) and 
+EntityFramework (as a store provider).
+
+Under the hood it uses [AspNet.Security.OpenIdConnect.Server](https://github.com/PinpointTownes/AspNet.Security.OpenIdConnect.Server) 
```
The sentence sounds weird (and the link should point to aspnet-contrib, not to my fork :smile:)

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,212 @@
+your application.
+
+Having multiple applications; maintainance of user databases and 
+registration/login workflows can easily became an administrative and 
```
nit: `became` -> `become`

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,212 @@
+login with existing Identity Providers.
+On the enterprise side, this would be ideally be one internal Identity Provider.
+
+Having an internal Idenity Provider provides you ability to use local password 
```
typo: `Idenity`

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,212 @@
+
+Having an internal Idenity Provider provides you ability to use local password 
+authentication, control the information that is exposed to the client applications
+and more generally, you can control who access your API.
```
`access` -> `can access`

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,212 @@
+
+### Configuration & Options.
+
+OpenIddict have multiple options you can set to customize for your 
```
`have` -> `has`

Commit:
Update asos project url, grammar and typos.

PureKrome commented:
Thoughts (in order of reading the read.me) ... and of course, this is IMO ...

### Title
- Title header is too long. Split into two.
- Loose the emoji in the title (or sub-heading if you do that).

### Under the title - first quick summary
- Needs `dev` and `master` badges. Ie. nuget version and nuget download count (for each). MyGet for `dev` branch. (then use AppVeyor for CI/CD).
- Needs a screenie for the NuGet cli: `install-package hello.is.it.me.youre.looking.for-.6.9.691

### Sales pitch
- Too long.
- I would literally try and keep the first sentence as the main paragraph describing the service. 
- Split the other para's into a others .. like _Why would I use this_ and _What happens under the hood_"

### What's OpenID Connect ?
- I think this is not really important (just yet) and should be dropped lower in the readme.
- Reword: "_The simplistic approach is to create a local database for the users’ accounts and credentials._" to (something like) "_Most developers usually create a `Users` database table and have a `UserName` or `Email`, `Password` and (hopefully) `PasswordSalt` fields for storing a clients credentials._"
- This whole section feels ... a bit long. But maybe it's necessary though.

### Why an OpenID Connect Server?
- Ah! i jumped the gun a bit (above) with suggesting this to be earlier. I still stand by that, thought.
- This entire section made no sense to me :( I still don't _get it_. Some hard examples are needed here. For example: Website sells movie tickets. Accounts are needed to purchase a ticket, blah blah blah. OpenIdDict can offer this and do that.. blah blah. money shot!

### All the rest below this.
- The rest (IMO) are wiki items. Too much for a ReadMe. Move all the code stuff into a wiki with pages .. not a page, but pages...

### Support / Contrib / Lic
- perfect. end the read me these :)

### Questions I'm wondering about, but not answered?
- OK, so this is some website thingy. Can i use this in my own _existing_ website? or is this a stand alone thing?
- If i can use it in my existing website, what about my existing DB? will it `fuck.things.up`:tm: ?

kevinchalet commented:
@PureKrome thanks for this great feedback! :clap: 

@Bartmax if you want to add the AppVeyor/Travis badges in this PR:

AppVeyor:

```
[![Build status](https://ci.appveyor.com/api/projects/status/d0d8git3o6lqkvbm?svg=true)](https://ci.appveyor.com/project/aspnet-contrib/core)
```

Travis:

```
[![Build status](https://travis-ci.org/openiddict/core.svg)](https://travis-ci.org/openiddict/core)
```

kevinchalet commented:
@Bartmax do you want to include the recommendations made by @PureKrome before we merge this PR? :stuck_out_tongue:

Bartmax commented:
ill make at least some in the next 2 days. let's wait for them before merge.

Commit:
React to Pure Krome suggestions.

Commit:
Fixed heading formatting.

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,102 @@
+
+### Why an OpenID Connect Server?
+
+With an OpenID Conenct Server you can manage all your users using local 
```
There's a little typo here: `OpenID Conenct Server` -> `OpenID Connect Server`.

Commit:
link to configuration and options in wiki.
fixed some typo
few rewording.

Commit:
fixed typo.

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,106 @@
+EntityFramework (as an optional store provider).
+
+Adding it to your existing application allows you to register clients and 
+serve them authenticate tokens.
```
`authenticate tokens` -> `authentication tokens`?

PureKrome replied:
`authenticate` tokens or `authenticaTION` tokens? (please don't throw stones at me if I've got this wrong, i always mess up my words...)

kevinchalet commented on README.md:
```diff
@@ -0,0 +1,106 @@
+
+Nightly builds can now be found on the [aspnet-contrib](https://github.com/aspnet-contrib) MyGet repository https://www.myget.org/F/aspnet-contrib/api/v3/index.json.
+
+To use OpenIddict Server you need to include OpenIddict as a dependency in your project.json:
```
`OpenIddict Server`?

PureKrome replied:
`To use....`  whoa... hold up a sec.

I think this needs to have a list summary of the steps needed to use this nuget package, before any code and text.

eg.

Steps to setup your webapp
1. install-package
2. modify `json.settings`
3. Wire up middleware in `ConfigureServices` (in `startup.cs`)

and now we can go into some code, etc...

Bascially, i need to get an overall feel about how complex it is (via the number of steps), to use this in my app.

Looking at the readme text below this point, i'm not sure if this is _everything_ or just the _hardest_ parts, of the steps.

Remember, i'm blond..... i need to be spoon fed.

PureKrome commented on README.md:
```diff
@@ -0,0 +1,106 @@
+AppVeyor: [![Build status](https://ci.appveyor.com/api/projects/status/d0d8git3o6lqkvbm?svg=true)](https://ci.appveyor.com/project/aspnet-contrib/core)
+
+Travis: [![Build status](https://travis-ci.org/openiddict/core.svg)](https://travis-ci.org/openiddict/core)
+
```
Include current NuGet version badges.

kevinchalet replied:
FYI, there's no official release on NuGet.org yet. I'd really like to upload an alpha version for RC1 though.

PureKrome commented on README.md:
```diff
@@ -0,0 +1,106 @@
+### Why an OpenID Connect Server?
+
+With an OpenID Connect Server you can manage all your users using local 
+password or an external identity provider management for all your applications 
```
Current: `or an external identity provider management`
Suggestion: `or an external identity provider (e.g. Login with Facebook) for all your...`

PureKrome commented on README.md:
```diff
@@ -0,0 +1,106 @@
+password or an external identity provider management for all your applications 
+in one central place, with the power to control who can access your API and 
+the information that is exposed to each client. 
+
```
A screen shot here might be worth 'a money shot'...

kevinchalet replied:
The native UI needs a large rework (it's the "basic" UI I used for ASOS samples):

![image](https://cloud.githubusercontent.com/assets/6998306/10988233/d9026712-843a-11e5-8ff0-e7addffd727b.png)

PureKrome commented on README.md:
```diff
@@ -0,0 +1,106 @@
+
+### How does it work?
+
+OpenIddict, by default, leverages the use of Identity (for user management) and 
```
Link `Identity` and `EF` to their respective GH repo's.

PureKrome commented on README.md:
```diff
@@ -0,0 +1,106 @@
+
+Under the hood it uses [AspNet.Security.OpenIdConnect.Server](https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server) 
+middleware that works with any standards-compliant OAuth 2.0/OpenID Connect 
+client including the official OpenID Connect client middleware 
```
Add link to the GH repo...

kevinchalet commented:
Merged, with the changes suggested by @PureKrome: https://github.com/openiddict/core/commit/ae38ba119360ae2d30f34f15bb952d55793a9a49

@PureKrome thanks for the review! :clap:
